src/lib/comms/sol-socket-dtls-impl-tinydtls.c

```diff
@@ -708,14 +796,50 @@ sol_socket_default_dtls_new(const struct sol_socket_options *options)
         goto err;
     }
 
-    dtls_set_handler(s->context, &dtls_handler);
+    s->handler.write = write_encrypted;
+    s->handler.read = call_user_read_cb;
+    s->handler.event = handle_dtls_event;
+
+    for (i = 0; i < opts.cipher_suites_len; i++) {
+        switch (opts.cipher_suites[i]) {
+        case SOL_SOCKET_DTLS_CIPHER_PSK_AES128_CCM8:
+            SOL_DBG("Adding get_psk_info callback to %p handler", &s->handler);
+            s->handler.get_psk_info = get_psk_info;
+            break;
+        case SOL_SOCKET_DTLS_CIPHER_ECDHE_ECDSA_AES128_CCM8:
+            SOL_DBG("Adding get_ecdsa_* callbacks to %p handler", &s->handler);
+            s->handler.get_ecdsa_key = get_ecdsa_key;
+            s->handler.verify_ecdsa_key = verify_ecdsa_key;
+            break;
+        case SOL_SOCKET_DTLS_CIPHER_ECDH_ANON_AES128_CBC_SHA256:
+#ifdef DTLS_EXTRAS
+            //FIXME: What should be done here?
```

Comment:
Well, I will remove the `#ifdef` then, since I will just say it's unsupported and return in both `#ifdef` branches.